lang/gjs/src/gtk3/astalify.ts

```diff
@@ -43,7 +43,7 @@ function setProp(obj: any, prop: string, value: any) {
 
 export default function astalify<
     C extends { new(...args: any[]): Gtk.Widget },
->(cls: C) {
+>(cls: C, childrenSetter: ((children: Gtk.Widget[], self: InstanceType<C>) => void) | null = null) {
```

Comment:
an API suggestion: let widgets extending `astalify` declare how they want children to be set. 
I'm planning to wrap adwaita, it'd be so much easier to work with. 
plus, this can be extended in future with more controlling things like `removeChild`, `addChild` etc if Bindings api will allow for that.

Reply:
@Aylur would love your opinion on this :)